spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/convert/DefaultDbRefResolver.java

```diff
@@ -106,7 +107,7 @@ public DBRef createDbRef(org.springframework.data.mongodb.core.mapping.DBRef ann
 	 */
 	@Override
 	public DBObject fetch(DBRef dbRef) {
-		return ReflectiveDBRefResolver.fetch(mongoDbFactory.getDb(), dbRef);
+		return ReflectiveDBRefResolver.fetch(MongoClientVersion.isMongo3Driver() ? mongoDbFactory.getDb() : null, dbRef);
```

Comment:
We need to move the entire logic of the if-clause up into `DefaultDbRefResolver` if we want to drop 2.x support so it's gonna change anyway. So yeah please go ahead and keep everything in `ReflectiveDbRefResolver.fetch(…)`.